analyzers/example/example_base/interface.py

```diff
@@ -0,0 +1,110 @@
+from analyzer_interface import (
+    AnalyzerInput,
+    AnalyzerInterface,
+    AnalyzerOutput,
+    InputColumn,
+    OutputColumn,
+)
+
+
+
+interface = AnalyzerInterface(
+    # Should be globally unique.
+    id="__example__",
+
+    # We don't really use this yet, but specify something for now.
+    version="0.1.0",
+
+    # The name of the analyzer as shown on the UI.
+    name="Example Analyzer",
+
+    # These descriptions are shown to the user in the UI at some point during the
+    # analysis selection process.
+    short_description="Example Analyzer (Character Count)",
+    long_description="""
+This is an example analzyer that counts the number of characters in each message.
+  """,
+    input=AnalyzerInput(
+        columns=[
+            InputColumn(
+                # This is the column name that you will use in your data analysis
+                # code.
+                name="message_id",
+
+                # This is the human readable name that will be displayed in the
+                # user interface.
+                human_readable_name="Unique Message ID",
+
+                # Refer to the complete set of data types by following the
+                # type definition.
+                data_type="identifier",
+
+                # This is a description of the column that will be displayed in
+                # the user interface during column matching.
+                description="The unique identifier of the message",
+
+                # This name hints give the application a kind of soft heuristics
+                # to match the column to the right data. The user will be able to
+                # override the suggestion if it is incorrect.
+                #
+                # You don't need to provide all possible hints, but the more you
+                # provide, the better the suggestions will be.
+                name_hints=[
+                    "post",
+                    "message",
+                    "comment",
+                    "text",
+                    "retweet id",
+                    "tweet",
+                ],
+            ),
+            InputColumn(
+                name="message_text",
+                human_readable_name="Message Text",
+                data_type="text",
+                description="The text content of the message",
+                name_hints=[
+                    "message",
+                    "text",
+                    "comment",
+                    "post",
+                    "body",
+                    "content",
+                    "tweet",
+                ],
+            )
+        ]
+    ),
+    outputs=[
+        AnalyzerOutput(
```

Comment:
@soul-codes What if you have multiple analysis columns that you want to produce as output? Say for example, both character count and word count?

Reply:
Oh wait! would that go in the secondary analyzer?